### DISCUSSION TITLE
Show newly added tracked files with gix status

### DISCUSSION THREAD

**Discussion by jakewilliami** [GitoxideLabs/gitoxide#1680] [Q&A]
It seems that `gix status` doesn't show newly
```commandline
$ git status --short
 M Cargo.toml
A  profile.json
 M src/main.rs
?? src/something.js

$ gix status
 14:11:14 status done 21.0 files in 0.00s (12.7K files/s)
  M Cargo.toml
  M src/main.rs
  ? src/something.js

head -> index isn't implemented yet
```

Is this expected behaviour?  Is this what is meant by `head -> index isn't implemented yet`?  And if so, is this on the roadmap?

I am using the `gix` library so I wonder if there is a workaround that I am not able to find.  In the library, I am able to show _all_ tracked files:
```rust
repo
    .status(gix::progress::Discard)?
    .dirwalk_options(|cb| {
        cb.emit_tracked(true)
    })
    .into_index_worktree_iter([])?
    .flatten()
```

But this includes files that are tracked but haven't been recently modified or added (for example, `Cargo.lock`), and they seem to have the same status:
```
[src/status.rs:281:13] &item = DirectoryContents {
    entry: Entry {
        rela_path: "profile.json",
        status: Tracked,
        property: None,
        disk_kind: Some(
            File,
        ),
        index_kind: Some(
            File,
        ),
        pathspec_match: Some(
            Always,
        ),
    },
    collapsed_directory_status: None,
}

[src/status.rs:281:13] &item = DirectoryContents {
    entry: Entry {
        rela_path: "Cargo.lock",
        status: Tracked,
        property: None,
        disk_kind: Some(
            File,
        ),
        index_kind: Some(
            File,
        ),
        pathspec_match: Some(
            Always,
        ),
    },
    collapsed_directory_status: None,
}
```

**Comment by Byron** [comment]
> Is this expected behaviour? Is this what is meant by `head -> index isn't implemented yet`? And if so, is this on the roadmap?

Yes, this message describes the issue you are experiencing, it's expected, and it's on the roadmap, hopefully it can still happen this year or early next year.
It cannot be worked around either.

**Reply by jakewilliami** [reply]
Noted, thanks Sebastian
